frontend/catalyst/jax_extras/patches.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Jax extras module containing Jax patches  """
+
+# pylint: disable=too-many-arguments
+
+from __future__ import annotations
+
+import jax
+from jax._src.lax.slicing import (
+    _gather_shape_computation,
+    _is_sorted,
+    _no_duplicate_dims,
+    _rank,
+    _sorted_dims_in_range,
+)
+from jax.core import AbstractValue, Tracer, concrete_aval
+
+__all__ = (
+    "get_aval2",
+    "_no_clean_up_dead_vars",
+    "_gather_shape_rule_dynamic",
+)
+
+
+def get_aval2(x):
+    """An extended version of `jax.core.get_aval` which also accepts AbstractValues."""
+    # TODO: remove this patch when https://github.com/google/jax/pull/18579 is merged
+    if isinstance(x, AbstractValue):
+        return x
+    elif isinstance(x, Tracer):
+        return x.aval
+    else:
+        return concrete_aval(x)
+
+
+def _no_clean_up_dead_vars(_eqn, _env, _last_used):
+    """A stub to workaround the Jax ``KeyError 'a'`` bug during the lowering of Jaxpr programs to
+    MLIR with the dynamic API enabled."""
+    return None
+
+
+def _gather_shape_rule_dynamic(
```

Comment:
@grwlf did you open an issue for this upstream?

Reply:
Yes, this one https://github.com/google/jax/pull/19083

Reply:
Just add a reference to this PR in the code please :)